shell/components/ClusterProviderIconMenu.vue

```diff
@@ -0,0 +1,149 @@
+<script>
+export default {
+  props: {
+    cluster: {
+      type:     Object,
+      required: true,
+    },
+  },
+  computed: {
+    isEnabled() {
+      return !!this.cluster?.ready;
+    },
+  },
+  methods: {
+    smallIdentifier(input) {
+      if (this.cluster.badge?.iconText) {
+        return this.cluster.badge?.iconText;
+      }
+
+      if (!input) {
+        return '';
+      }
+
+      const segments = input.match(/([A-Z]?[a-z]+|\d+)/g);
+
+      if (!segments) return ''; // In case no valid segments are found
+
+      let result = '';
+
+      switch (segments.length) {
+      case 1:
+        // eslint-disable-next-line no-case-declarations
+        const word = segments[0];
+
+        result = `${ word[0] }${ word[Math.floor(word.length / 2)] }${ word[word.length - 1] }`;
+        break;
+      case 2:
+        result = `${ segments[0][0] }${ segments[1][0] }${ segments[1][segments[1].length - 1] }`;
+        break;
+      default:
+        result = segments.slice(0, 3).map((segment) => segment[0]).join('');
+      }
+
+      return result.toUpperCase();
+    },
```

Comment:
Maybe a comment about what is about? Is it about create acronymous of the label?
Also we should make a test for this.

Reply:
Yeah sounds good to me too 👍

Reply:
Ok I've seen it now in the E2E, kinda odd but at least is tested.